src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -383,19 +389,21 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
 
     @action
     public onGeneSelect(vertical:boolean, entrezGeneId:number) {
-        let targetSelection;
-        let otherSelection;
+        let targetSelection:AxisMenuSelection;
+        let otherSelection:AxisMenuSelection;
         if (vertical) {
             targetSelection = this.vertSelection;
             otherSelection = this.horzSelection;
         } else {
             targetSelection = this.horzSelection;
             otherSelection = this.vertSelection;
         }
-        targetSelection.entrezGeneId = entrezGeneId;
-        if (this.geneLock) {
-            otherSelection.entrezGeneId = entrezGeneId;
-        }
+        runInAction(()=>{
```

Comment:
this is already an action, so why why runInAction?

Reply:
is every set in an observable automatically an action?

Reply:
oh oops i got confused